pulsar-testclient/src/main/java/org/apache/pulsar/testclient/PerformanceProducer.java

```diff
@@ -325,6 +343,37 @@ public static void main(String[] args) throws Exception {
             printAggregatedStats();
         }));
 
+        if (arguments.createTopicPartitions > 0) {
+            PulsarAdminBuilder clientBuilder = PulsarAdmin.builder()
+                    .serviceHttpUrl(arguments.adminURL)
+                    .tlsTrustCertsFilePath(arguments.tlsTrustCertsFilePath);
+
+            if (isNotBlank(arguments.authPluginClassName)) {
+                clientBuilder.authentication(arguments.authPluginClassName, arguments.authParams);
+            }
+
+            if (arguments.tlsAllowInsecureConnection != null) {
+                clientBuilder.allowTlsInsecureConnection(arguments.tlsAllowInsecureConnection);
+            }
+
+            try (PulsarAdmin client = clientBuilder.build();) {
+                for (String topic : arguments.topics) {
+                    log.info("Creating partitioned topic {} with {} partitions", topic, arguments.createTopicPartitions);
+                    try {
+                        client.topics().createPartitionedTopic(topic, arguments.createTopicPartitions);
+                    } catch (PulsarAdminException.ConflictException alreadyExists) {
+                        log.debug("Topic "+topic+" already exists: " + alreadyExists);
+                        PartitionedTopicMetadata partitionedTopicMetadata = client.topics().getPartitionedTopicMetadata(topic);
+                        if (partitionedTopicMetadata.partitions != arguments.createTopicPartitions) {
+                            log.error("Topic {} already exists but it has a wrong number of partitions: {}, expecting {}",
+                                    topic, partitionedTopicMetadata.partitions, arguments.createTopicPartitions);
+                            System.exit(-1);
```

Comment:
why exit(-1) here? Why can we throw an exception?

Reply:
we are using System.exit(-1); all over this class.
I have filed #9865 in order to clean up this problem.

Reply:
@sijie I finally ended up in adding support for not calling System.exit.
I had to do so in order to allow tests to not crash the JVM